test/compiler/inference.jl

```diff
@@ -2660,3 +2660,8 @@ function symcmp36230(vec)
     return false
 end
 @test Base.return_types(symcmp36230, (Vector{Any},)) == Any[Bool]
+
+# Issue #36531, double varargs in abstract_iteration
+splat36531(args...) = args
+f36531(args...) = Core._apply_iterate(Base.iterate, splat36531, args...)
+@test @inferred(f36531(1,2,3)) == (1,2,3)
```

Comment:
This tests seems to pass after printer the error message, which might go unnoticed if it regresses.
Maybe something like
```suggestion
@test code_typed(f36531, Tuple{Vararg{Int}}) isa Vector
```

Reply:
I'll do both, run it with `@inferred` and the code typed to test correctness as well as absence of error.